src/main/java/com/session/service/client/SessionClientImpl.java

```diff
@@ -114,11 +117,8 @@ public Session getSessionByEmail(String email) throws SessionClientException {
         ZebedeeSession session = null;
 
         if (StringUtils.isNotEmpty(email)) {
-            String uri = format("/search?email={0}", email);
-            HttpGet httpGet = http.createHttpGet(host, uri);
-
             try {
-                session = http.doGet(httpGet, getSessionResponseHandler());
+                session = http.get(host, "/sessions/" + email, getSessionResponseHandler());
```

Comment:
Do the same `session == null` check afterwards

Reply:
Was going to get to the email bits in the next PR just pushing them method by method.